src/rgw/rgw_lc.cc

```diff
@@ -502,8 +491,8 @@ class LCObjsLister {
   }
 
   boost::optional<std::string> next_key_name() {
-    if (obj_iter == objs.end() ||
-	(obj_iter + 1) == objs.end()) {
+    if (obj_iter == list_results.objs.end() ||
+	(obj_iter + 1) == list_results.objs.end()) {
```

Comment:
the next_key_name() mechanism needs to still work--I think I wrote improved s3-tests cases for this, but we should be sure

Reply:
It should be good.